hang in ud_mlx5/test_ud.ca_md/1

http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3396/label=hpc-test-node2,worker=2/console
```
17:00:12 [ RUN      ] ud_mlx5/test_ud.ca_md/1
18:37:22 Build timed out (after 120 minutes). Marking the build as failed.
18:37:22 Build was aborted
18:37:22 TAP Reports Processing: START
18:37:22 Looking for TAP results report in workspace using pattern: **/*.tap
18:37:23 
make: *** [test] Terminated
```

--- a/test/gtest/uct/test_ud.cc
+++ b/test/gtest/uct/test_ud.cc
@@ -9,6 +9,7 @@
 extern "C" {
 #include <ucs/time/time.h>
 #include <ucs/datastruct/queue.h>
+#include <ucs/arch/bitops.h>
 #include <uct/ib/ud/base/ud_ep.h>
 };
 
@@ -514,7 +515,7 @@ UCS_TEST_P(test_ud, ca_ai) {
 UCS_TEST_P(test_ud, ca_md, "IB_TX_QUEUE_LEN=" UCS_PP_MAKE_STRING(UCT_UD_CA_MAX_WINDOW)) {
 
     ucs_status_t status;
-    int new_cwnd;
+    int prev_cwnd, new_cwnd;
     int i;
 
     if (RUNNING_ON_VALGRIND) {
@@ -541,17 +542,20 @@ UCS_TEST_P(test_ud, ca_md, "IB_TX_QUEUE_LEN=" UCS_PP_MAKE_STRING(UCT_UD_CA_MAX_W
 
     ep(m_e1)->tx.tx_hook = count_tx;
     do {
-        new_cwnd = ep(m_e1, 0)->ca.cwnd / UCT_UD_CA_MD_FACTOR;
+        prev_cwnd = ep(m_e1, 0)->ca.cwnd;
         tx_count = 0;
         do {
             progress();
-        } while (ep(m_e1, 0)->ca.cwnd != new_cwnd);
+        } while (ep(m_e1, 0)->ca.cwnd > (prev_cwnd / UCT_UD_CA_MD_FACTOR));
         short_progress_loop();
 
-        /* up to 3 additional ack_reqs per each resend */
+        new_cwnd = ep(m_e1, 0)->ca.cwnd;
         EXPECT_GE(tx_count, new_cwnd - 1);
-        EXPECT_LE(tx_count, new_cwnd - 1 + 3);
-        EXPECT_EQ(ep(m_e1, 0)->ca.cwnd, new_cwnd);
+        if (new_cwnd > UCT_UD_CA_MIN_WINDOW) {
+           /* up to 3 additional ack_reqs per each resend */
+           EXPECT_LE(tx_count, (new_cwnd - prev_cwnd) +
+                               3 * ucs_ilog2(prev_cwnd/new_cwnd));
+	}
 
     } while (ep(m_e1, 0)->ca.cwnd > UCT_UD_CA_MIN_WINDOW);
 }

--- a/test/gtest/uct/test_ud.cc
+++ b/test/gtest/uct/test_ud.cc
@@ -553,8 +553,8 @@ UCS_TEST_P(test_ud, ca_md, "IB_TX_QUEUE_LEN=" UCS_PP_MAKE_STRING(UCT_UD_CA_MAX_W
         EXPECT_GE(tx_count, new_cwnd - 1);
         if (new_cwnd > UCT_UD_CA_MIN_WINDOW) {
            /* up to 3 additional ack_reqs per each resend */
-           EXPECT_LE(tx_count, (new_cwnd - prev_cwnd) +
-                               3 * ucs_ilog2(prev_cwnd/new_cwnd));
+           EXPECT_LE(tx_count, (prev_cwnd - new_cwnd) +
+                               (int)(3 * ucs_ilog2(prev_cwnd/new_cwnd)));
 	}
 
     } while (ep(m_e1, 0)->ca.cwnd > UCT_UD_CA_MIN_WINDOW);

